src/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Win32;
+
+namespace System.Net.WebSockets
+{
+    internal static class WebSocketValidate
+    {
+        internal const int MaxControlFramePayloadLength = 123;
+
+        private const int CloseStatusCodeAbort = 1006;
+        private const int CloseStatusCodeFailedTLSHandshake = 1015;
+        private const int InvalidCloseStatusCodesFrom = 0;
+        private const int InvalidCloseStatusCodesTo = 999;
+        private const string Separators = "()<>@,;:\\\"/[]?={} ";
+
+        internal static void ValidateSubprotocol(string subProtocol)
+        {
+            if (string.IsNullOrWhiteSpace(subProtocol))
+            {
+                throw new ArgumentException(SR.net_WebSockets_InvalidEmptySubProtocol, "subProtocol");
+            }
+
+            string invalidChar = null;
+            int i = 0;
+            while (i < subProtocol.Length)
+            {
+                char ch = subProtocol[i];
+                if (ch < 0x21 || ch > 0x7e)
```

Comment:
It'd be nice if there was a comment or something indicating what these boundaries were (I had to look them up in an ascii table).